### PR TITLE
feat: Added LR Finder for TensorFlow scripts

### LINKED ISSUE #478: [references] Add a LR Finder in the training scripts
For specific architectures, the learning rate that we generally use seems to yield various performances. We need a more consistent method to select an appropriate value for it.

Adding an option to select the learning rate would be quite valuable!
- [x] PyTorch support (#703)
- [ ] TensorFlow support (#747)

### PR DESCRIPTION
**Pull request by fg-mindee**
This PR introduces the following modifications:
- fixes output node sizing for mobile nets in TF & PyTorch (encountered problems with the script in char classification)
- adds LR finder option for all supported tasks in TensorFlow

Closes #478

Any feedback is welcome!

### CODE REVIEW

**Review by charlesmindee — APPROVED** [review]
Thanks!
